### PR TITLE
Allow importing electrodes to an existing project

### LINKED ISSUE #222: Allow importing electrodes to an existing project
## Description

Currently, the only way of importing electrodes into a project is during the import of a segmentation, by placing an `electrodes.dat` file into the same folder with the segmentation to be imported, and then going to **Menu tool > Import > Import data to a new project**, and choosing the `import_segmentation.zef` script in the same folder. It would be beneficial, if electrodes could be imported into an existing project after the segmentation has already been imported, so that different electrode configurations could be used in simulations with the same head model more effortlessly.

## Expected behavior

One would expect there to be a working button **Menu tool > Import > Import electrodes**, that would allow one to select a file of electrode positions and labels, which would then get imported to the project.

## Context

| Name | Value |
| --- | --- |
| **Version** | f958de80 Merge pull request #193 from sampsapursiainen/192-fs2zefsh-improvements |
| **OS** | Any and all |
| **Shell** | Any and all |

### PR DESCRIPTION
**Pull request by SeSodesa**
- Adds the package `import` with the functions `electrodes_from_dat` and `electrodes_from_csv`, which read electrode positions, labels and possible inner and outer radii and impedances from a given `.dat` or `.csv` text file.
- Adds an <kbd>Import electrodes</kbd> button to **Menu tool > Import**, with an attached callback `gui.menu_tool.import_electrodes_callback`.
- Closes #222 .

### CODE REVIEW

**Comment by SeSodesa** [comment]
This now seems to somewhat work on my fork of the project ([link]), at least with the electrode file you provided, but the question still remains whether the fields that the electrode labels and numerical data are saved to are the desired ones. Currently they are set as `zef.s2_name_list` and `zef.sensors`. Also, I have not yet tested this with data that contains complete electrode model (CEM) fields. A bit more documentation still needs to be added, as well.

I'll upload the data to this repo as well, once I have a machine with the proper permissions set up available to me (my work computer). I couldn't be bothered to set up the required expiring tokens on our workstations for multiple remotes.

[link]: https://github.com/SeSodesa/zeffiro_interface/tree/222-allow-importing-electrodes-to-an-existing-project

File: +import/electrodes_from_csv.m [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    column_titles = string ( electrode_table.Properties.VariableNames ) ;
+
+    [ n_of_rows, ~ ] = size ( electrode_table ) ;
+
+    % Preallocate outputs.
+
+    electrode_data = zeros ( n_of_rows, 6 ) ;
+
+    electrode_labels = repmat ( kwargs.MISSING_LABEL, n_of_rows, 1 ) ;
```
Comment (mshavliuk) [inline +import/electrodes_from_csv.m:62]: So eventually, if the names of the sensors were not specified, all of them will be named "N/A"? In such case I would prefer the previous way of defining names for them, by enumerating from index 1 and so on.

**Review by mshavliuk — CHANGES_REQUESTED** [review]
The new import module looks great and documented nicely, great work 👏

For some reason the electrodes do not appear in the Segmentation tool at the moment, but as I understand it's still WIP.

Also, I've noticed one minor bug: when I'm closing the file picker for the electrodes import it complains about the wrong file type:

https://github.com/sampsapursiainen/zeffiro_interface/assets/6589665/ed223031-76ed-47e7-ad0a-d210e8c464fa

File: +import/electrodes_from_csv.m (hunk 1 above)
Comment (SeSodesa) [reply +import/electrodes_from_csv.m]: This should now be resolved. The electrodes with missing labels will be named `SN` ,where `N` is the row of the electrode in the input file, if no label is given.

Comment (SeSodesa) [reply +import/electrodes_from_csv.m]: Also, the case of a user pressing "Cancel" in the file selector should also be handled now.

**Pull request merged**